### ISSUE TITLE
Update event-driven-evolution.md for evolveRL design (rename + new contract)

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#2625]
## Summary
Rewrite `docs/event-driven-evolution.md` so it documents the renamed `Creature.evolveRL()` API, the class-shaped adapter contract, default termination guards, seed cadence, per-creature episode averaging, and the opt-in statistics schedule decided in #2624. This doc is the contract that the implementation sub-issues are measured against.

## What Needs to Be Done

1. **Rename throughout** — every `evolveEnv` reference in `docs/event-driven-evolution.md` becomes `evolveRL`. Add a one-paragraph note that `evolveEnv` was the design-phase name (RFC #2610) and was renamed before any implementation shipped, so there is no migration path to worry about.
2. **Justify `evolveRL` as the canonical name** — replace the old "Why evolveEnv" section with a new section that names the alternatives considered (`evolveEnv`, `evolveOnline`, `evolveInteractive`) and explains why `evolveRL` wins: it explicitly signals reinforcement-learning semantics to readers familiar with Gym/Gymnasium and PyTorch-NEAT, rather than implying generic "environment" interaction.
3. **Replace the structural-typed `EpisodeAdapter` interface with a class-shaped contract**:
   - **Abstract methods** (MUST override): `reset(rngSeed: number)`, `step(action: A): { observation, reward, terminated, truncated, info? }`, action and observation shape declarations (`actionShape`, `observationShape` or equivalent).
   - **Overridable guard methods** (MAY override): `maxSteps()` (default `5000`), `wallClockMs()` (default `60_000`).
   - **Final / library-owned**: the `runEpisode()` wrapper that drives the loop. Caller does not implement this.
   - Use Gym/Gymnasium return shape (`obs, reward, terminated, truncated, info`) so users coming from those libraries recognise it immediately.
4. **Document the default termination guards**:
   - Wall-clock cap = **60 seconds per episode** (primary guard — library cannot know step cost in advance).
   - Max-steps cap = **5000 steps per episode**.
   - Whichever fires first marks the episode as **truncated** (not terminated). Both are overridable per adapter. Snake hitting a wall returns `terminated = true`; pressing-left-forever hits the wall-clock guard and returns `truncated = true`.
5. **Document the seed cadence**:
   - Within a single generation, **all creatures play the same N seeded episodes** (default `N = 3`).
   - The seed set **rotates per generation** (gen `g+1` derives a fresh seed set deterministically from the parent seed) to avoid overfitting one map.
   - A fixed seed (no rotation) is opt-in via an option and intended only for tests / regression — never the default in real evolution.
6. **Document the per-creature averaging** — `episodesPerCreature = 3` by default; per-creature fitness is the **mean return** across those episodes for noise reduction. Show how this maps to `error` for the existing `targetError` / plateau machinery.
7. **Document the opt-in statistics schedule**:
   - Off by default (performance / memory).
   - When enabled, milestones follow the **geometric schedule** `1, 2, 5, 10, 20, 50, 100, 200, 500, 1000, …`.
   - Per-milestone payload: best creature's score / return, best creature's neuron count and synapse count, mean episode length (steps) across the generation, generation wall-clock time.
8. **Cross-reference #2612** — the worker contract there must align with the new adapter shape. Add a short note that `evolveRL` accepts a serialisable adapter description (import URL + JSON config) to enable per-episode workers per Option A in #2612.
9. **Acceptance criteria block** — replace the existing "evolveEnv shipped" criteria with the renamed criteria and add tickets for the new behaviours (termination guards, seed rotation, opt-in stats, geometric milestone schedule).

## Acceptance Criteria

- [ ] `docs/event-driven-evolution.md` no longer mentions `evolveEnv` outside the rename note.
- [ ] The `evolveRL` name is justified explicitly against `evolveEnv` / `evolveOnline` / `evolveInteractive`.
- [ ] The class-shaped adapter contract is fully documented (abstract / overridable / final tiers).
- [ ] Default termination guards (60s wall-clock, 5000 steps, truncated semantics) are documented.
- [ ] Seed-cadence rules (per-generation rotation, fixed-seed opt-in) are documented.
- [ ] `episodesPerCreature = 3` averaging is documented.
- [ ] Opt-in statistics schedule (geometric: 1, 2, 5, 10, 20, 50, 100, 200, 500, 1000) and payload are documented.
- [ ] `docs/REINFORCEMENT_LEARNING.md` cross-references the renamed API where appropriate.
- [ ] `./quality.sh` passes (Markdown lint + link check).

## Dependencies
None — design refresh of an existing RFC document.

## Diagram
```mermaid
classDiagram
    class EpisodeAdapter~S,A~ {
      <<abstract>>
      +reset(rngSeed)*
      +step(action)* obs,reward,term,trunc
      +observationShape*
      +actionShape*
      +maxSteps() default 5000
      +wallClockMs() default 60000
    }
    class EpisodeRunner {
      <<final>>
      +runEpisode(adapter, creature, seed)
    }
    EpisodeRunner --> EpisodeAdapter : drives
```

## Context
Part of #2624. Foundation document; all implementation sub-issues reference this doc as the contract. Replaces the `evolveEnv` spec from RFC #2610.

**Comment by stservice** [comment]
Pull request https://github.com/stSoftwareAU/NEAT-AI/pull/2631 has been created to address this issue.

**Comment by stservice** [comment]
Automatically closed — PR #2631 has already been merged for this issue.

---

Closed-PR self-healing (Issue #787)
